### PR TITLE
Included Framework Capabilities, Formatting and Simple tests to begin…

### PR DESCRIPTION
**Pull request by mrrajan**
This PR includes,
 - Python packages and dependencies to support playwright python execution
 - Script to install and configure virtual environment 
 - Formatting/Editor configuration and Pytest HTML report
 - A simple playwright python script which capable of running on trustification staging environment. In order to run the script, please register a user and pass the credentials from CLI.
 - The README and CONTRIBUTING markdown file updated for quick start up. Reference and learning docs would be included in further commits.

Results - [Testresults](https://github.com/trustification/spog-test/files/14179051/testrun-2024-02-06-17-21-49.zip)

### CODE REVIEW

File: .isort.cfg
Comment (ikanias) [inline .isort.cfg:1]: @mrrajan can you please explain what 'multi_line_output', 'force_grid_wrap' mean here? Does  'force_grid_wrap' means the max number of lines written in as a wrap?

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# Contributing
+In order to contribute,
+ - Fork this repo
+ - Create a Branch on your forked repo
+ - Developer features and Create a Git Commit
```
Comment (ikanias) [inline CONTRIBUTING.md]: Line 5 is not very clear. I think maybe it should say: 'Add your changes and then create a Git commit'

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+ - Push your changes
+ - Create a Pull Request
+
+# Formatting
+Before creating a Pull Request, make sure to scripts are formatted and aligned. The formatting rules are controlled by python packages,
```
Comment (ikanias) [inline CONTRIBUTING.md]: Please fix '...make sure to scripts are formatted' to 'make sure that the scripts are formatted'

Comment (ikanias) [inline CONTRIBUTING.md]: Please remove the ',' in the end of line 10

File: README.md [hunk 3]
```diff
@@ -1,8 +1,30 @@
...
+
+## Quick Start
+1. Clone the repository 
+
+```
+git clone https://github.com/trustification/spog-test.git
+cd spog-test
+```
+
+2. The shell script [run_spog_tests.sh](run_spog_tests.sh) is capable of setting up the virtual environment and installing the required packages to execute the Automation scripts. At the end of the execution the HTML reports are stored into `results` directory. To run the script, the user should pass trustification application URL, registered Username and Password.
```
Comment (ikanias) [inline README.md:13]: Maybe we should emphasize that the username and password are 'Trustification' username and password? WDYT?

File: setup.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
+from setuptools import find_packages, setup
+
+with open('requirements.txt') as f:
+    requirements=f.read().splitlines()
+
+setup(
+    name="spog-ui-tests",
+    version="0.0.1",
+    author="RHTC QE team",
```
Comment (ikanias) [inline setup.py]: @mrrajan maybe we should change it to author="RHTAP QE team"?

File: setup.py [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    author_email="rhtc-qe@redhat.com",
+    description="UI Tests for Trustification SPoG",
+    url="https://github.com/trustification/spog-test.git",
+    pakages=find_packages(),
```
Comment (ikanias) [inline setup.py]: @mrrajan please fix 'pakages' to 'packages'

Comment (ikanias) [inline setup.py]: @mrrajan how does the find_packages() function work? According to https://www.geeksforgeeks.org/what-is-setup-py-in-python/ the 'packages' parameter should include the package you need to install and the 'install_requires' parameter should only contain the package's dependencies.

File: tests/conftest.py [hunk 6]
```diff
@@ -0,0 +1,24 @@
+import pytest
```
Comment (ikanias) [inline tests/conftest.py:1]: @mrrajan is this something that must be run before each test? And does the login screen is the same today with the branded UI as when you wrote this?

**Review by ikanias — CHANGES_REQUESTED** [review]
Can you please answer the questions I added?

File: .isort.cfg
Comment (mrrajan) [reply .isort.cfg:1]: @ikanias these two are options for isort configuration. You can find the use of these here -  https://pycqa.github.io/isort/docs/configuration/options.html#force-grid-wrap
To put it simple, it acts on the imports - all the imports aligned in a separate line even though it is from same package.

File: README.md (hunk 3 above)
Comment (mrrajan) [reply README.md:13]: Hmm, it is mentioned that the user should pass trustification application URL, registered Username and Password. Please let me know if this needs to be specified as 'Trustification' username and 'Trustification' password.

File: setup.py (hunk 4 above)
Comment (mrrajan) [reply setup.py]: Ah, good catch. The script was created way back before.

File: setup.py (hunk 5 above)
Comment (mrrajan) [reply setup.py]: find_packages() simply retrieves all the functions from the root directory and it is useful for the package and distribution. I am not sure about the article though, may be they want to cherry pick the packages which are only to be installed on the client. 
install_requires is for more basic packages which needs to be installed. Here I point it to requirements file. We could change it in the future for specific dependencies in future.

File: tests/conftest.py (hunk 6 above)
Comment (mrrajan) [reply tests/conftest.py:1]: I am not sure which one you are referring to, is it `import pytest`? or the function `setup_before_test`? 
Regarding the UI, I am not pretty sure - this code was written 3 months back. I will give it a try and let you know.

Comment (mrrajan) [reply tests/conftest.py:1]: Just executed against non-aws installation, it works well. Please note, the login screen is designed for keycloak. In order to handle AWS congnito we might need to have configuration and scripts in place.

**Pull request merged**